simulation/traffic_simulator/test/src/traffic_lights/test_traffic_light.cpp

```diff
@@ -723,39 +726,68 @@ TEST(TrafficLight, Bulb_trafficLightMessageConversion)
   }
 }
 
-/**
- * @note Test basic functionality. Test function behavior when called with invalid name.
- */
-TEST(TrafficLight, Color_make_wrong)
+class TrafficLightTest : public testing::Test
 {
-  EXPECT_THROW(traffic_simulator::TrafficLight::Color::make("wrong_color"), common::SyntaxError);
-}
+protected:
+  TrafficLightTest()
+  : map_manager(
+      ament_index_cpp::get_package_share_directory("traffic_simulator") + "/map/lanelet2_map.osm",
+      geographic_msgs::build<geographic_msgs::msg::GeoPoint>()
+        .latitude(35.61836750154)
+        .longitude(139.78066608243)
+        .altitude(0.0))
+  {
+  }
+  hdmap_utils::HdMapUtils map_manager;
+};
 
 /**
- * @note Test basic functionality. Test function behavior when called with invalid name. 
+ * @note test if function correctly determines if a given bulb
+ * is in the bulbs vector given a Color, Status, Shape triple.
  */
-TEST(TrafficLight, Shape_make_wrong)
+TEST_F(TrafficLightTest, contains_colorStatusShape)
 {
-  EXPECT_THROW(traffic_simulator::TrafficLight::Shape::make("wrong_shape"), common::SyntaxError);
-}
+  using TrafficLight = traffic_simulator::TrafficLight;
+  using Color = TrafficLight::Color;
+  using Status = TrafficLight::Status;
+  using Shape = TrafficLight::Shape;
 
-/**
- * @note Test basic functionality. Test function behavior when called with invalid name.
- */
-TEST(TrafficLight, Status_make_wrong)
-{
-  EXPECT_THROW(traffic_simulator::TrafficLight::Status::make("wrong_status"), common::SyntaxError);
+  {
+    auto traffic_light = TrafficLight(34802, map_manager);
+
+    traffic_light.bulbs.emplace(Color::red, Status::flashing, Shape::circle);
+    traffic_light.bulbs.emplace(Color::green, Status::solid_on, Shape::right);
+
+    ASSERT_TRUE(traffic_light.bulbs.size() == static_cast<std::size_t>(2));
+    EXPECT_TRUE(traffic_light.contains(Color::red, Status::flashing, Shape::circle));
+    EXPECT_TRUE(traffic_light.contains(Color::green, Status::solid_on, Shape::right));
+  }
+
+  {
+    auto traffic_light = TrafficLight(34802, map_manager);
+
+    traffic_light.bulbs.emplace("red flashing circle");
+    traffic_light.bulbs.emplace("green solidOn right");
+
+    EXPECT_TRUE(traffic_light.contains(Color::red, Status::flashing, Shape::circle));
+    EXPECT_TRUE(traffic_light.contains(Color::green, Status::solid_on, Shape::right));
+  }
 }
 
 /**
- * @note Test basic functionality. Test function behavior when called with invalid name.
+ * @note Test function behavior with a valid string.
  */
-TEST(TrafficLight, Bulb_make_wrong)
+TEST_F(TrafficLightTest, set_valid)
 {
-  using Bulb = traffic_simulator::TrafficLight::Bulb;
+  using TrafficLight = traffic_simulator::TrafficLight;
+  using Color = TrafficLight::Color;
+  using Status = TrafficLight::Status;
+  using Shape = TrafficLight::Shape;
```

Comment:
I think this should be moved to the fixture class

Reply:
I will just move usings to the top of the file